### PR TITLE
Fix flaky console test

### PR DESCRIPTION
**Pull request by blindpirate**
By asserting `containsString` instead of `startsWith`.

### CODE REVIEW

**Comment by blindpirate** [comment]
Fixes https://github.com/gradle/gradle-private/issues/3685

**Comment by blindpirate** [comment]
@bot-gradle test this

File: subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/AbstractConsoleBuildResultFunctionalTest.groovy [hunk 1]
```diff
@@ -112,7 +114,7 @@ BUILD SUCCESSFUL in [ \\dms]+
 
         and:
         // Ensure the failure is a location that the fixtures can see
-        failure.assertHasDescription("Execution failed for task ':broken'")
+        failure.assertThatDescription(containsString("Execution failed for task ':broken'"))
```
Comment (wolfs) [inline subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/AbstractConsoleBuildResultFunctionalTest.groovy:117]: 🤔 I think it makes sense to use `contains` only for the debug log level. For the other log levels, nothing should slip into to description. WDYT?

**Review by wolfs — APPROVED** [review]

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**